controller.go

```diff
@@ -522,6 +522,8 @@ func (c *controller) NewSandbox(containerID string, options ...SandboxOption) (S
 
 	heap.Init(&sb.endpoints)
 
+	sb.resolver = NewResolver(sb)
```

Comment:
This can be moved inside the `startResolver` function since this is not anyways needed until there is a need to start resolver. That way you can do proper cleanup of resolver if there is any failure in `startResolver`